sklearn/manifold/setup.py

```diff
@@ -14,6 +16,16 @@ def configuration(parent_package="", top_path=None):
                          include_dirs=[numpy.get_include()],
                          libraries=libraries,
                          extra_compile_args=["-O3"])
+    cblas_libs, blas_info = get_blas_info()
+    eca = blas_info.pop('extra_compile_args', [])
+    eca.append("-O4")
+    config.add_extension("_barnes_hut_tsne",
+                         libraries=cblas_libs,
+                         sources=["_barnes_hut_tsne.c"],
```

Comment:
I think you need to add utils here. What was the motivation of putting one function in a different file? Do you think it will be more widely useful?